walnut/README.md

```diff
@@ -1,4 +1,76 @@
 # `walnut` - the O² workflow administration and linting utility
 
 The O² **w**orkflow **a**dministration and **l**i**n**ting **ut**ility is a command line program for workflow configuration
-and import validation.
\ No newline at end of file
+and import validation.
+
+## Configuration file
+
+You can check the local `walnut` configuration with
+```bash
+$ walnut about
+```
+
+## Using `walnut`
+
+At any step, you can type `$ walnut help <subcommand>` to get information on what you can do and how, for example:
+
+```bash
+$ walnut help 
+walnut is a command line program for interacting with the O² Workflow Administation and Linting Utility.
+
+For more information on the available commands, see the individual documentation for each command.
+
+Usage:
+  walnut [command]
+
+Available Commands:
+  check       check the file passed against a specified schema.
+  convert     Converts a DPL Dump to the required formats.
+  help        Help about any command
+
+Flags:
+      --config string   optional configuration file for walnut (default $HOME/.config/walnut/settings.yaml)
+  -h, --help            help for walnut
+  -t, --toggle          Help message for toggle
+
+Use "walnut [command] --help" for more information about a command.
+```
+
+### Validating a file
+
+To check if a file conforms to the pre-defined schemas, run the `check` command: 
+```bash
+$ walnut check
+```
+Example usage:
+```bash
+$ walnut check --filename dump.json --format workflow 
+```
+
+A successful validation will return no output while any errors will cause the program to exit.
+
+#### SEE ALSO
+
+* [walnut check](./doc/walnut_check.md)
+
+
+### Validating a file
+
+The goal of `walnut` is to convert O² generated dumps to the required number of task templates and workflow templates. The
```

Comment:
Since AliECS is part of O², I think this should be reworded to "DPL generated dumps" or "O²/DPL generated dumps".